lib/flapjack/gateways/jabber.rb

```diff
@@ -422,6 +423,65 @@ def interpreter(command_raw, from)
             msg = "that doesn't seem to be a valid pattern - /#{pattern}/"
           end
         end
+          
+        when /^(?:maint )?entities\s+\/(.+)\/(?:\s*(.*?)(?:\s*start-in:.*?(\w+.*?))?(?:\s*start-at:.*?(\w+.*?))?(?:\s*duration:.*?(\w+.*?))?)$/i
+          entity_pattern   = $1.strip
+          comment          = $2 ? $2.strip : nil
+          start_in         = $3 ? $3.strip : nil
+          start_at         = $4 ? $4.strip : nil
+          duration_str     = $5 ? $5.strip : '1 hour'
+          maint_duration   = ChronicDuration.parse(duration_str)
+          entity_names     = Flapjack::Data::Entity.find_all_name_matching(entity_pattern, :redis => @redis)
+
+          if comment.nil? || (comment.length == 0)
+            comment = "#{from}: Set via chatbot"
+          else
+            comment = "#{from}: #{comment}"
+          end
+
+          if start_in 
+            maint_start = Time.now.to_i + ChronicDuration.parse(start_in)
+          elsif start_at
+            maint_start = Chronic.parse(start_at).to_i
+          else
+            maint_start = Time.now.to_i
+          end
+
+          if entity_names
+            number_found = entity_names.length
+            case
+            when number_found == 0
+              msg = "found no entities matching /#{pattern}/"
+            when number_found >= 1
+              entities = entity_names.map {|name|
+                Flapjack::Data::Entity.find_by_name(name, :redis => @redis)
+              }.compact.inject({}) {|memo, entity|
+                memo[entity] = entity.check_list.sort
+                memo
+              }
+              if entities.length >= 1
+                entities.each_pair do |entity,check_list|
+                  check_list.each do |check|
+                    entity_check = Flapjack::Data::EntityCheck.for_entity(entity, check, :redis => @redis)
+                    entity_check.create_scheduled_maintenance(
+                      maint_start,
+                      maint_duration,
+                      :summary => comment,
+                      )
+                    entity_check.update_current_scheduled_maintenance
```

Comment:
While this is true, this refactor can wait until v2 (where a lot has already changed anyway)